torch/_inductor/utils.py

```diff
@@ -1251,10 +1251,19 @@ def use_cpp_packed_gemm_template(layout, mat1, mat2, mat2_transposed=False):
         num_threads=parallel_num_threads(),
     )
 
+    def is_last_dim_stride1(x):
+        if isinstance(x.layout, ir.FixedLayout):
```

Comment:
If this is failing, should we be calling freeze_layout?  Flexible layouts are allowed to change dynamically.  If you have a FlexibleLayout you can force the last dim to be stride=1

Reply:
Thanks for the suggestion. I've updated the code. Could you help take another look?